nipype/interfaces/niftyreg/reg.py

```diff
@@ -0,0 +1,403 @@
+    ln_val = traits.Range(low=0, argstr='-ln %d',
+                          desc='Number of resolution levels to create')
+    lp_val = traits.Range(low=0, argstr='-lp %d',
+                          desc='Number of resolution levels to perform')
+    nopy_flag = traits.Bool(desc='Do not use the multiresolution approach',
+                            argstr='-nopy')
+    noconj_flag = traits.Bool(desc='Use simple GD optimization',
+                              argstr='-noConj')
+    desc = 'Add perturbation steps after each optimization step'
+    pert_val = traits.Range(low=0, desc=desc, argstr='-pert %d')
+
+    # F3d2 options
+    vel_flag = traits.Bool(desc='Use velocity field integration',
+                           argstr='-vel')
+    fmask_file = File(exists=True,
+                      desc='Floating image mask',
+                      argstr='-fmask %s')
+
+    # Other options
+    desc = 'Kernel width for smoothing the metric gradient'
+    smooth_grad_val = traits.Float(desc=desc, argstr='-smoothGrad %f')
+    # Padding value
+    pad_val = traits.Float(desc='Padding value', argstr='-pad %f')
+    # verbosity off
+    verbosity_off_flag = traits.Bool(argstr='-voff',
+                                     desc='Turn off verbose output')
+
+    # Output CPP image file
+    cpp_file = File(name_source=['flo_file'],
+                    name_template='%s_cpp.nii.gz',
+                    desc='The output CPP file',
+                    argstr='-cpp %s')
+    # Output warped image file
+    res_file = File(name_source=['flo_file'],
+                    name_template='%s_res.nii.gz',
+                    desc='The output resampled image',
+                    argstr='-res %s')
+
+
+class RegF3DOutputSpec(TraitedSpec):
+    """ Output Spec for RegF3D. """
+    cpp_file = File(desc='The output CPP file')
+    res_file = File(desc='The output resampled image')
+    invcpp_file = File(desc='The output inverse CPP file')
+    invres_file = File(desc='The output inverse res file')
+    desc = 'Output string in the format for reg_average'
+    avg_output = traits.String(desc=desc)
+
+
+class RegF3D(NiftyRegCommand):
+    """Interface for executable reg_f3d from NiftyReg platform.
+
+    Fast Free-Form Deformation (F3D) algorithm for non-rigid registration.
+    Initially based on Modat et al., "Fast Free-Form Deformation using
+    graphics processing units", CMPB, 2010
+
+    For source code, see https://cmiclab.cs.ucl.ac.uk/mmodat/niftyreg
+
+    Examples
+    --------
+    >>> from nipype.interfaces import niftyreg
+    >>> node = niftyreg.RegF3D()
+    >>> node.inputs.ref_file = 'im1.nii'
+    >>> node.inputs.flo_file = 'im2.nii'
+    >>> node.inputs.rmask_file = 'mask.nii'
+    >>> node.inputs.omp_core_val = 4
+    >>> node.cmdline  # doctest: +ALLOW_UNICODE
+    'reg_f3d -cpp im2_cpp.nii.gz -flo im2.nii -omp 4 -ref im1.nii \
+-res im2_res.nii.gz -rmask mask.nii'
+
+    """
+    _cmd = get_custom_path('reg_f3d')
```

Comment:
@oesteban - i thought that this was an interesting approach and one that could be extended to support containers as well. let's merge this PR in before this current release (tomorrow) and then think about the few additional pieces (namesource and this custom path updates) for the next release.